File: docs/swingjs/j2s/javax/microedition/rms/RecordEnumerationImpl.js
(function(){var P$=Clazz.newPackage("javax.microedition.rms"),I$=[[0,'java.util.ArrayList','java.util.Collections','java.util.Comparator']],I$0=I$[0],$I$=function(i,n,m){return m?$I$(i)[n].apply(null,m):((i=(I$[i]||(I$[i]=Clazz.load(I$0[i])))),!n&&i.$load$&&Clazz.load(i,2),i)};
/*c*/var C$=Clazz.newClass(P$, "RecordEnumerationImpl", null, null, 'javax.microedition.rms.RecordEnumeration');

C$.$clinit$=2;

Clazz.newMeth(C$, '$init$', function () {
},1);

C$.$fields$=[['I',['cursor'],'O',['store','javax.microedition.rms.RecordStore','ids','java.util.List']]]

Clazz.newMeth(C$, 'c$$javax_microedition_rms_RecordStore$javax_microedition_rms_RecordFilter$javax_microedition_rms_RecordComparator$Z',  function (store, filter, comparator, keepUpdated) {
;C$.$init$.apply(this);
this.store=store;
this.ids=Clazz.new_($I$(1,1));
var n=store.getNumRecords$();
for (var i=1; i <= n; i++) {
var rec=store.getRecord$I(i);
if (filter == null  || filter.matches$BA(rec) ) {
this.ids.add$O(Integer.valueOf$I(i));
}}
if (comparator != null ) {
$I$(2,"sort$java_util_List$java_util_Comparator",[this.ids, $I$(3,"comparing$java_util_function_Function$java_util_Comparator",[((P$.RecordEnumerationImpl$lambda1||
(function(){/*m*/var C$=Clazz.newClass(P$, "RecordEnumerationImpl$lambda1", function(){Clazz.newInstance(this, arguments[0],1,C$);}, null, 'java.util.function.Function', 1);

C$.$clinit$=2;

Clazz.newMeth(C$, '$init$', function () {
},1);
/*lambda_E*/
Clazz.newMeth(C$, ['apply$Integer','apply$O'],  function (id) { return (this.$finals$.store.getRecord$I((id).$c()));});
})()
), Clazz.new_(P$.RecordEnumerationImpl$lambda1.$init$,[this, {store:store}])), ((P$.RecordEnumerationImpl$lambda2||
(function(){/*m*/var C$=Clazz.newClass(P$, "RecordEnumerationImpl$lambda2", function(){Clazz.newInstance(this, arguments[0],1,C$);}, null, 'java.util.Comparator', 1);

C$.$clinit$=2;

Clazz.newMeth(C$, '$init$', function () {
},1);
/*lambda_E*/
Clazz.newMeth(C$, ['compare$BA$BA','compare$O$O'],  function (a, b) { return (this.$finals$.comparator.compare$BA$BA(a, b));});
})()
), Clazz.new_(P$.RecordEnumerationImpl$lambda2.$init$,[this, {comparator:comparator}]))])]);
}if (keepUpdated) {
}this.cursor=0;
}, 1);

Clazz.newMeth(C$, 'numRecords$',  function () {
return this.ids.size$();
});

Clazz.newMeth(C$, 'hasNextElement$',  function () {
return this.cursor < this.ids.size$();
});

Clazz.newMeth(C$, 'nextRecord$',  function () {
if (!this.hasNextElement$()) {
throw Clazz.new_(Clazz.load('javax.microedition.rms.InvalidRecordIDException'));
}var id=(this.ids.get$I(this.cursor++)).$c();
return this.store.getRecord$I(id);
});

Clazz.newMeth(C$, 'nextRecordId$',  function () {
if (!this.hasNextElement$()) {
throw Clazz.new_(Clazz.load('javax.microedition.rms.InvalidRecordIDException'));
}return (this.ids.get$I(this.cursor)).$c();
});

Clazz.newMeth(C$, 'reset$',  function () {
this.cursor=0;
});

Clazz.newMeth(C$, 'rebuild$',  function () {
this.cursor=0;
});

Clazz.newMeth(C$, 'destroy$',  function () {
this.ids.clear$();
});

Clazz.newMeth(C$);
})();
;Clazz.setTVer('3.3.1-v5');//Created 2026-06-25 15:29:03 Java2ScriptVisitor version 3.3.1-v5 net.sf.j2s.core.jar version 3.3.1-v5

File: src/javax/microedition/rms/RecordEnumerationImpl.java
package javax.microedition.rms;

import java.util.ArrayList;
import java.util.Collections;
import java.util.Comparator;
import java.util.List;

final class RecordEnumerationImpl implements RecordEnumeration {
  private final RecordStore store;
  private final List<Integer> ids;
  private int cursor;

  RecordEnumerationImpl(
      RecordStore store, RecordFilter filter, RecordComparator comparator, boolean keepUpdated) {
    this.store = store;
    this.ids = new ArrayList<>();
    int n = store.getNumRecords();
    for (int i = 1; i <= n; i++) {
      byte[] rec = store.getRecord(i);
      if (filter == null || filter.matches(rec)) {
        ids.add(i);
      }
    }
    if (comparator != null) {
      Collections.sort(
          ids,
          Comparator.comparing(
              (Integer id) -> store.getRecord(id),
              (a, b) -> comparator.compare(a, b)));
    }
    if (keepUpdated) {
      // Optional MIDP behaviour; AquaDig passes false.
    }
    cursor = 0;
  }

  @Override
  public int numRecords() {
    return ids.size();
  }

  @Override
  public boolean hasNextElement() {
    return cursor < ids.size();
  }

  @Override
  public byte[] nextRecord() throws InvalidRecordIDException {
    if (!hasNextElement()) {
      throw new InvalidRecordIDException();
    }
    int id = ids.get(cursor++);
    return store.getRecord(id);
  }

  @Override
  public int nextRecordId() throws InvalidRecordIDException {
    if (!hasNextElement()) {
      throw new InvalidRecordIDException();
    }
    return ids.get(cursor);
  }

  @Override
  public void reset() {
    cursor = 0;
  }

  @Override
  public void rebuild() {
    cursor = 0;
  }

  @Override
  public void destroy() {
    ids.clear();
  }
}
